src/methods/simulate.jl

```diff
@@ -42,16 +43,19 @@ function simulate{T}(sys::LtiSystem{Val{T},Val{:disc}}, tspan;
 
   f     = (t,x,dx)->sys(t,x,dx,input)
   simvar= SimType(initial, zeros(numoutputs(sys)), zeros(numinputs(sys)))
+  temp = zeros(simvar.x)
+  f(0.0, simvar, temp)  # ensure first step is correct
+  simvar.x = temp 
 
-  prob  = DiscreteProblem(f, simvar, tspan)
   dt    = samplingtime(sys)
+  prob  = DiscreteProblem(f, simvar, tspan) # extra time step, see below
   sln   = OrdinaryDiffEq.solve(prob, FunctionMap(scale_by_time=false);
     kwargs..., dt = samplingtime(sys))
```

Comment:
`dt = samplingtime(sys)` above is not used. we should remove it. or, we can write here as,
```julia
sln = OrdinaryDiffEq.solve(prob, FunctionMap(scale_by_time=false); kwargs..., dt = dt)
```